src/test/java/bio/terra/service/filedata/azure/tables/TableDirectoryDaoConnectedTest.java

```diff
@@ -86,23 +84,44 @@ public void cleanup() throws Exception {
     // Should already be deleted
     for (String entry : directoryEntriesToCleanup) {
       try {
-        tableDirectoryDao.deleteDirectoryEntry(tableServiceClient, entry);
+        tableDirectoryDao.deleteDirectoryEntry(
+            tableServiceClient, datasetId, DATASET_TABLE.toTableName(), entry);
       } catch (Exception ex) {
         logger.debug("Directory entry either already deleted or unable to delete {}", entry, ex);
       }
     }
-    if (tableName != null) {
+    if (snapshotTableName != null) {
       try {
-        TableClient tableClient = tableServiceClient.getTableClient(tableName);
+        TableClient tableClient = tableServiceClient.getTableClient(snapshotTableName);
         tableClient.deleteTable();
       } catch (Exception ex) {
-        logger.error("Unable to delete table {}", tableName, ex);
+        logger.error("Unable to delete table {}", snapshotTableName, ex);
       }
     }
 
     connectedOperations.teardown();
   }
 
+  @Test
+  public void testStoreTopDirectory() {
+    tableDirectoryDao.storeTopDirectory(tableServiceClient, snapshotId, dataset.getName());
+
+    snapshotTableName = SNAPSHOT_TABLE.toTableName(snapshotId);
+    int count = getTableEntryCount(tableServiceClient, snapshotTableName);
+    assertThat("Store top directory should add two entries to snapshot table.", count, equalTo(2));
+
+    // get directories to confirm the correct ones are added
+    List<String> directories = new ArrayList();
+    directories.add("/");
+    directories.add("/_dr_");
```

Comment:
The `storeTopDirectory` method will also create a directory for `dataset.getName()` if it doesn't exist so you can add that into the directories to check:

```suggestion
    directories.add("/_dr_");
    directories.add(dataset.getName());
```

Reply:
Oh this only creates two directory entries so I think the first two paths are getting the same record

Reply:
Oh!! Thank you for catching this! I've changed the batch retrieve to also check for unique paths - I assume we wouldn't want to return duplicates.

Reply:
No problem! Checking for unique paths is a good call!!